### PR TITLE
Fix api reference docs

### PR DESCRIPTION
**Pull request by typhoonzero**
Some document polish work.

machi 2018-06-15 12-38-17
machi 2018-06-15 12-38-03
machi 2018-06-15 12-35-12
machi 2018-06-15 12-34-42
machi 2018-06-15 12-33-43
machi 2018-06-15 12-29-04
machi 2018-06-15 12-27-59
machi 2018-06-15 12-27-21
machi 2018-06-15 12-22-34
machi 2018-06-15 12-21-59
machi 2018-06-15 12-21-17
machi 2018-06-15 12-20-50

### CODE REVIEW

**Comment by reyoung** [comment]
I just polish `templatedoc()` a little bit to generate better documentation. Make it support inline math and correct the google style comments. See #11308.

Also, as discussed with @shanyi15 yesterday, we decided to use a new repo to manage documentation of fluid for several reasons.

1. We need cross hyperlink between API-reference, API-Guide and User story. In Paddle, we separate API, use case into many sphinx projects( around 10 sphinx projects before... T_T ). They cannot be cross-link well.
2. It is hard to build documentation before. Use `cmake` to drive `sphinx-build` is tricky. We can separate them into two projects. It is easy to complete i18n later if we use a standard sphinx project structure.

Current, this repo(https://github.com/reyoung/FluidDoc) is in very early stage and flux. However, it is very simple to build a documentation site follow the readme. 

You can install your developing version of paddle, and `make html`. The documentation will be generated well at `build/html` directory.


The develop branch of this documentation will be generated to `readthedocs.org`. http://fluiddoc.readthedocs.io/zh_CN/develop/ . It uses our latest release paddle to generate.

**Comment by typhoonzero** [comment]
@reyoung Thanks for polishing `templatedoc`, I was looking into the format problem using local compiled server. Glad that you already make a fix.

File: paddle/fluid/framework/details/fuse_vars_op_handle.cc [hunk 1]
```diff
@@ -42,7 +42,7 @@ void FuseVarsOpHandle::RunImpl() {
     out_t->ShareDataWith(out_tensor->Slice(s, s + numel));
     s += numel;
   }
-  this->RunAndRecordEvent([this] {});
+  this->RunAndRecordEvent([] {});
```
Comment (typhoonzero) [inline paddle/fluid/framework/details/fuse_vars_op_handle.cc]: This line is intended to fix build on macos

**Review by skylarch — DISMISSED** [review]

**Comment by skylarch** [comment]
dynamic_lstm解释公式变量的部分改为列表形式，和之前的保持一致。

**Review by skylarch — CHANGES_REQUESTED** [review]
request changes as commented

**Comment by skylarch** [comment]
@typhoonzero 
现在@reyoung 老师可能对注释内容有新的建议，API注释标准暂时待定，之前的comment可以先不做修改。

**Comment by skylarch** [comment]
@typhoonzero 
comment已根据最新标准更新，可以开始做修改。

**Comment by typhoonzero** [comment]
@skylarch Added screen shot of the document preview.

**Comment by skylarch** [comment]
@typhoonzero 

chunk_eval: 
缺标题；
缺examples；
“see below for details"指的是哪一部分呢？这些可能是有问题的${chunk_scheme_comment}，${num_chunk_types_comment}，${excluded_chunk_types_comment}

iou_similarity
标题格式除介词外首字母大写，无句号，加粗

ctc_greedy_decoder
截图不全

ListenAndServ
标题格式除介词外首字母大写，无句号，加粗
缺return

autoincreased_step_counter
缺例子

conv2d_transpose
标题格式除介词外首字母大写，无句号，加粗
截图不全

roi_pool
标题格式除介词外首字母大写，无句号，加粗
例子或截图不全

crf_decoding
例子

scale
例子

cos_sim
标题格式除介词外首字母大写，无句号，加粗
例子

**Comment by reyoung** [comment]
default
default
default
default
default

**Review by reyoung — DISMISSED** [review]
Excellent

**Review by skylarch — DISMISSED** [review]

**Review by reyoung — APPROVED** [review]
Cool

**Pull request merged**